app/packages/state/src/hooks/useSetView.ts

```diff
@@ -104,41 +91,38 @@ const useSetView = (
               : viewOrUpdater;
           commit({
             variables: {
-              subscription: subscription,
-              session: session,
+              subscription,
+              session,
               view: value,
               datasetName: dataset.name,
               form: patch
-                ? {
-                    filters: snapshot.getLoadable(filters).contents,
-                    sampleIds: [
-                      ...snapshot.getLoadable(selectedSamples).contents,
-                    ],
-                    labels: snapshot.getLoadable(selectedLabelList).contents,
-                    extended: snapshot.getLoadable(fos.extendedStages).contents,
-                    addStages,
-                    slice: selectSlice
-                      ? snapshot.getLoadable(resolvedGroupSlice(false)).contents
-                      : null,
-                  }
+                ? snapshot.getLoadable(
+                    viewStateForm({
+                      addStages: addStages ? JSON.stringify(addStages) : null,
+                      modal: false,
+                      selectSlice,
+                    })
+                  ).contents
                 : {},
-
-              changingSavedView: changingSavedView,
-              savedViewSlug: savedViewSlug,
-              viewName: viewName,
+              changingSavedView,
+              savedViewSlug,
+              viewName,
             },
             onError,
             onCompleted: ({ setView: { dataset, view: value } }) => {
               const isDesktop = isElectron();
-              if (router.history.location.state) {
+              if (
+                router.history.location.state &&
+                router.history.location.state.state
+              ) {
                 const newState = {
-                  ...router.history.location.state,
+                  ...router.history.location.state.state,
```

Comment:
@manivoxel51 I changed this because before we were creating increasingly nested state each time this code was triggered